runtest.py

```diff
@@ -831,12 +846,13 @@ def run(self):
         sys.stdout.write("\t" + "\n\t".join(paths) + "\n")
 
 # save the fails to a file
-if fail and args.error_log:
-    paths = [x.path for x in fail]
-    #print(f"DEBUG: Writing fails to {args.error_log}")
+if args.error_log:
```

Comment:
What's the functional difference here?
Before it wouldn't open the file if there were no failures, and now it opens the file but doesn't write if there're no failures?

Reply:
always writes the file if a file is configured (which it is by default).  the effect is that if there were no fails it will be empty, where before previous contents stuck around if there were no new fails to cause it to be rewritten.  This probably isn't the perfect solution either, but it was cheap.

Reply:
Good enough for sure!